js/dydx.js

```diff
@@ -0,0 +1,939 @@
+        const stopPrice = this.safeNumber (order, 'trigger_price');
+        const result = {
+            'info': order,
+            'id': id,
+            'clientOrderId': clientId,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'lastTradeTimestamp': undefined,
+            'symbol': symbol,
+            'type': type,
+            'timeInForce': timeInForce,
+            'postOnly': undefined,
+            'side': side,
+            'price': price,
+            'stopPrice': stopPrice,
+            'amount': amount,
+            'cost': cost,
+            'average': undefined,
+            'filled': filled,
+            'remaining': remaining,
+            'status': status,
+            'trades': undefined,
+        };
+        return result;
+    }
+
+    parseDepositAddress (depositAddress, currency = undefined) {
+        const address = this.safeString (depositAddress, 'address');
+        const currencyId = this.safeString (depositAddress, 'currency');
+        const code = this.safeCurrencyCode (currencyId);
+        return {
+            'info': depositAddress,
+            'code': code,
+            'address': address,
+        };
+    }
+
+    parseOHLCV (ohlcv, market = undefined) {
+        //
+        // {
+        //   "startedAt": "2021-01-05T00:00:00.000Z",
+        //   "market": "BTC-USD",
+        //   "resolution": "1DAY",
+        //   "low": "40000",
+        //   "high": "45000",
+        //   "open": "45000",
+        //   "close": "40000",
+        //   "baseTokenVolume": "1.002",
+        //   "trades": "3",
+        //   "usdVolume": "45085",
+        //   "startingOpenInterest": "28"
+        // },
+        //
+        const timestamp = this.safeInteger (ohlcv, 'startedAt');
+        const open = this.safeNumber (ohlcv, 'open');
+        const high = this.safeNumber (ohlcv, 'high');
+        const low = this.safeNumber (ohlcv, 'low');
+        const close = this.safeNumber (ohlcv, 'close');
+        const volume = this.safeNumber (ohlcv, 'usdVolume');
+        return [ timestamp, open, high, low, close, volume ];
+    }
+
+    sign (path, api = 'private', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        const version = this.safeString (this.options, 'version', 'v3');
+        let url = this.urls['api']['private'] + '/' + version + '/' + path;
+        const keys = Object.keys (params);
+        const length = keys.length;
+        let query = undefined;
+        if (length > 0) {
+            if (method === 'GET') {
+                query = this.urlencode (params);
+                url = url + '?' + query;
+            } else {
+                body = this.json (params);
+            }
+        }
+        headers = {
+            'Content-Type': 'application/json',
+        };
+        if (this.privateKey !== undefined) {
+            headers['DYDX-SIGNATURE'] = this.privateKey;
```

Comment:
this doesn't look right

Reply:
@frosty00 I know. I wanted to first get some feedback on EIP-712-compliant Ethereum signatures before adding a signature there. Anyway, I've copied what was done on js/idex.js. 
Does this look right?